dataframe-arrow/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/ArrowKtTest.kt

```diff
@@ -273,4 +285,87 @@ internal class ArrowKtTest {
+        )
+
+        val dataFrame = dataFrameOf(
+            "ts_nano" to dates,
+            "ts_micro" to dates,
+            "ts_milli" to dates,
+            "ts_sec" to dates)
+
+        DataFrame.readArrowFeather(writeArrowTimestamp(dates)) shouldBe dataFrame
+        DataFrame.readArrowIPC(writeArrowTimestamp(dates,true)) shouldBe dataFrame
+
+    }
+
+    private fun  writeArrowTimestamp(dates:List<LocalDateTime>,streaming:Boolean=false) :ByteArray {
+        RootAllocator().use { allocator ->
+            val timeStampMilli = Field(
+                "ts_milli",
+                FieldType.nullable(ArrowType.Timestamp(TimeUnit.MILLISECOND, null)),
+                null
+            )
+
+            val timeStampMicro = Field(
+                "ts_micro",
+                FieldType.nullable(ArrowType.Timestamp(TimeUnit.MICROSECOND, null)),
+                null
+            )
+
+            val timeStampNano = Field(
+                "ts_nano",
+                FieldType.nullable(ArrowType.Timestamp(TimeUnit.NANOSECOND, null)),
+                null
+            )
+
+            val timeStampSec = Field(
+                "ts_sec",
+                FieldType.nullable(ArrowType.Timestamp(TimeUnit.SECOND, null)),
+                null
+            )
+            val schemaTimeStamp = Schema(
+                listOf(timeStampNano,timeStampMicro,timeStampMilli,timeStampSec)
+            )
+            VectorSchemaRoot.create(schemaTimeStamp, allocator).use { vectorSchemaRoot ->
+                val timeStampMilliVector = vectorSchemaRoot.getVector("ts_milli") as TimeStampMilliVector
+                val timeStampNanoVector = vectorSchemaRoot.getVector("ts_nano") as TimeStampNanoVector
+                val timeStampMicroVector = vectorSchemaRoot.getVector("ts_micro") as TimeStampMicroVector
+                val timeStampSecVector = vectorSchemaRoot.getVector("ts_sec") as TimeStampSecVector
+                timeStampMilliVector.allocateNew(dates.size)
+                timeStampNanoVector.allocateNew(dates.size)
+                timeStampMicroVector.allocateNew(dates.size)
+                timeStampSecVector.allocateNew(dates.size)
+
+
+                dates.forEachIndexed { index, localDateTime ->
+                    val instant = localDateTime.toInstant(ZoneOffset.UTC)
+                    timeStampNanoVector[index] = instant.toEpochMilli() * 1000000L + instant.nano
+                    timeStampMicroVector[index] = instant.toEpochMilli() * 1000L
+                    timeStampMilliVector[index] = instant.toEpochMilli()
+                    timeStampSecVector[index] = instant.toEpochMilli() / 1000
+                }
+                vectorSchemaRoot.setRowCount(dates.size)
+                val bos = ByteArrayOutputStream()
+                bos.use { out ->
+                    val arrowWriter = if(streaming){
+                        ArrowStreamWriter(vectorSchemaRoot, null, Channels.newChannel(out))
+                    }else{
```

Comment:
please lint your files. You can check the result with `gradlew lintKotlin`

Reply:
`gradlew lintKotlinTest`  runs also successfully and same for a full build `gradlew clean build` ...
Does the JVM (I'm using Java 17) version can impact linter result ?  

```
 ./gradlew lintKotlinTest                                                                                                               ✔  30877  15:49:57

> Configure project :
Current DataFrame version: 0.13.0-dev

> Task :core:processKDocsMain
Doc Processor is running!
Running processor: nl.jolanrensen.docProcessor.defaultProcessors.IncludeDocProcessor
Running processor: nl.jolanrensen.docProcessor.defaultProcessors.IncludeFileDocProcessor
Running processor: nl.jolanrensen.docProcessor.defaultProcessors.IncludeArgDocProcessor
Running processor: nl.jolanrensen.docProcessor.defaultProcessors.CommentDocProcessor
Running processor: nl.jolanrensen.docProcessor.defaultProcessors.SampleDocProcessor

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.3-rc-2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 40s
27 actionable tasks: 2 executed, 25 up-to-date
```

Reply:
oh that's interesting, it looks like we're missing the kotlinter plugin outside of the :core and :plugins module...

I made an [issue](https://github.com/Kotlin/dataframe/issues/520) for it :) and I'll try to enable it for all modules right away. I'll let you know when it's added so you can rebase :)

Reply:
Ok, by the way I finally ran kotlin linter and formatter by adding (locally) `id("org.jmailen.kotlinter")` in plugins section and the same `kotlinter` config (as core module) to build.gragle.kts in the module
Some code that is not mine has been also formatted 
Would you like I update PR with the formatted/linted code  or wait fix for #520 ?

Reply:
I think it's best to wait otherwise we'll get more merge conflicts.